File: backend-csharp/DTOs/Responses/SessionResponse.cs
namespace Picture2Text.Api.DTOs.Responses;

/// <summary>
/// 會話資訊
/// </summary>
public class SessionInfo
{
    /// <summary>
    /// 會話 ID
    /// </summary>
    public string SessionId { get; set; } = string.Empty;

    /// <summary>
    /// 裝置名稱
    /// </summary>
    public string? DeviceName { get; set; }

    /// <summary>
    /// IP 地址
    /// </summary>
    public string? IpAddress { get; set; }

    /// <summary>
    /// 登入時間
    /// </summary>
    public DateTime LoginAt { get; set; }

    /// <summary>
    /// 最後活躍時間
    /// </summary>
    public DateTime LastActivityAt { get; set; }

    /// <summary>
    /// 是否為當前會話
    /// </summary>
    public bool IsCurrent { get; set; }

    /// <summary>
    /// 過期時間
    /// </summary>
    public DateTime ExpiresAt { get; set; }
}

/// <summary>
/// 會話列表回應
/// </summary>
public class SessionListResponse : ApiResponse<List<SessionInfo>>
{
}

File: backend-nestjs/src/dto/response/session.response.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class SessionInfo {
  @ApiProperty({
    description: 'Session ID',
    example: '550e8400-e29b-41d4-a716-446655440000',
  })
  SessionId: string;

  @ApiPropertyOptional({
    description: '裝置名稱',
    example: 'Chrome on Windows',
  })
  DeviceName?: string;

  @ApiPropertyOptional({
    description: 'IP 位址',
    example: '192.168.1.1',
  })
  IpAddress?: string;

  @ApiProperty({
    description: '登入時間',
    example: '2024-01-15T10:00:00.000Z',
  })
  LoginAt: Date;

  @ApiProperty({
    description: '最後活動時間',
    example: '2024-01-15T10:30:00.000Z',
  })
  LastActivityAt: Date;

  @ApiProperty({
    description: '過期時間',
    example: '2024-01-22T10:00:00.000Z',
  })
  ExpiresAt: Date;

  @ApiProperty({
    description: '是否為當前 Session',
    example: true,
  })
  IsCurrent: boolean;
}

export class SessionListResponse {
  @ApiProperty()
  Code: number;

  @ApiProperty()
  Message: string;

  @ApiProperty({ type: [SessionInfo] })
  Data: SessionInfo[];
}
